Doc/library/struct.rst

```diff
@@ -144,12 +151,10 @@ following table:
 
 If the first character is not one of these, ``'@'`` is assumed.
 
-Native byte order is big-endian or little-endian, depending on the host
-system. For example, Intel x86 and AMD64 (x86-64) are little-endian;
-IBM z and most legacy architectures are big-endian;
-and ARM, RISC-V and IBM Power feature switchable endianness
-(bi-endian, though the former two are nearly always little-endian in practice).
-Use ``sys.byteorder`` to check the endianness of your system.
+Native byte order is big-endian or little-endian, depending on the
+host system. For example, Intel x86, AMD64 (x86-64), and Apple M1 are
+little-endian; IBM z and most legacy architectures are big-endian.
+Use :func:`sys.byteorder` to check the endianness of your system.
```

Comment:
> I do not think we should drop the elaboration about switchable architectures - it further emphasises how variable "native mode" may be.

I disagree. It was inaccurate information already that just confused matters. It is not the job of Python's docs to try and keep up with the continual pile of technical details about specific low level platform capabilities.

There is no such thing as a switchable architecture from the point of the user. No supported operating system allows choice. It can never be changed at process runtime. Thus claiming there is something bi-endian and switchable is not useful information for someone to have. We similarly don't bother to tell people about middle and mixed endian.

What is useful is that people should call sys.byteorder() to find out if they ever have a need to know. The docs do that. When using the struct module APIs properly, code never has a reason to know the underlying platforms endianness. The point of the API is to abstract that.

Reply:
Next edit will change ``:func:`` to ``:data:``. Beginner's mistake on my part.

I'm going to side with Greg on the switchable endianness and leave it out, at least for now. (Maybe it deserves a footnote?) Still, trying to include potentially interesting (but ultimately not useful) computer architecture details seems counterproductive.

@gpshead, can you elaborate on this statement?

> When using the struct module APIs properly, code never has a reason to know the underlying platforms endianness. 

It seems one certainly needs to know who you are talking to and what the agreed upon endianness is, if for no other reason than to select the appropriate prefix character.

While I'm thinking about it, in the current crop of examples, only ``@``, ``<``, and ``>`` are used. Should the documentation motivate the use of ``=`` or ``!``? I have a difficult time envisioning a real use for ``=``. Perhaps an example would be useful. I do understand why one might want ``!``, but only so the programmer doesn't necessarily need to know that network byte order is big endian.